elyra/pipeline/component.py

```diff
@@ -119,39 +110,39 @@ def __init__(self, ref: str, name: str, type: str, value: str, description: str,
         self._required = required
 
     @property
-    def ref(self):
+    def ref(self) -> str:
         return self._ref
 
     @property
-    def name(self):
+    def name(self) -> str:
         return self._name
 
     @property
-    def type(self):
+    def type(self) -> str:
         return self._type
 
     @property
-    def value(self):
+    def value(self) -> str:
         return self._value
 
     @property
-    def description(self):
+    def description(self) -> str:
         return self._description
 
     @property
-    def control(self):
+    def control(self) -> str:
         return self._control
 
     @property
-    def control_id(self):
+    def control_id(self) -> str:
         return self._control_id
 
     @property
-    def items(self):
+    def items(self) -> List[str]:
         return self._items
 
     @property
-    def required(self):
+    def required(self) -> bool:
         if self._required:
```

Comment:
we could simplify this to `return bool(self._required)` since that will also treat `None` values as False.